gateway-js/src/index.ts

```diff
@@ -116,11 +120,41 @@ function isCsdlConfig(config: GatewayConfig): config is CsdlGatewayConfig {
   return 'csdl' in config;
 }
 
+// A manually managed config means the user has provided a function which
+// handles providing service definitions to the gateway.
+function isManuallyManagedConfig(
+  config: GatewayConfig,
+): config is ManuallyManagedGatewayConfig {
+  return 'experimental_updateServiceDefinitions' in config;
+}
+
+// Managed config strictly means managed by Studio
 function isManagedConfig(
   config: GatewayConfig,
 ): config is ManagedGatewayConfig {
   return (
-    !isRemoteConfig(config) && !isLocalConfig(config) && !isCsdlConfig(config)
+    !isRemoteConfig(config) &&
+    !isLocalConfig(config) &&
+    !isCsdlConfig(config) &&
+    !isManuallyManagedConfig(config)
+  );
+}
+
+// A static config is one which loads synchronously on start and never updates
+function isStaticConfig(
+  config: GatewayConfig,
+): config is LocalGatewayConfig | CsdlGatewayConfig {
+  return isLocalConfig(config) || isCsdlConfig(config);
+}
+
+// A dynamic config is one which loads asynchronously and (can) update via polling
+function isDynamicConfig(
+  config: GatewayConfig,
+): config is RemoteGatewayConfig | ManagedGatewayConfig {
```

Comment:
This looks wrong: it should include `| ManuallyManagedConfig`, right? (Maybe this is an argument against my previous suggestion since this mistake would be less obvious if there was a `type DynamicConfig` somewhere else.)

Reply:
af14289